### PR TITLE
docs(samples): Update table.getRows and dataset access samples

### PR DESCRIPTION
**Pull request by steffnay**
- Updates samples for query auto-pagination and paginating through table rows.

- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)

 🦕

### CODE REVIEW

File: samples/browseTable.js
Comment (JustinBeckwith) [inline samples/browseTable.js:59]: This has to be inside of the region tag, or the code will never be invoked when the customer tries it.

File: samples/queryPagination.js
Comment (JustinBeckwith) [inline samples/queryPagination.js:70]: Same issue, needs to be inside the region tag

**Review by JustinBeckwith — CHANGES_REQUESTED** [review]

File: samples/browseTable.js
Comment (tswast) [inline samples/browseTable.js:41]: It'd be helpful to show what operations are possible on rows. Are they indexable by number or column name, for example.

(We don't actually do that in Python https://github.com/googleapis/python-bigquery/blob/9d43d2073dc88140ae69e6778551d140430e410d/samples/browse_table_data.py but if I did it again, I'd prefer listing rows from a sample table with a well-known schema.

File: samples/queryPagination.js
Comment (tswast) [inline samples/queryPagination.js:62]: Is it possible to paginate over the results without a destination table?

File: samples/queryPagination.js
Comment (tswast) [inline samples/queryPagination.js:54]: I recall that the Node.js library used to download all result rows in one of these methods. Is that no longer the case?

File: samples/browseTable.js
Comment (steffnay) [reply samples/browseTable.js:41]: I think rewriting this sample to be able to do that will introduce complexity that decreases its overall quality. This library ultimately infers the table's project ID from the instance of the client, so creating a reference to a public datasets table in order to call getRows() would require showing users to manually reset the projectId, which would likely create a lot of confusion. Alternately, creating a table using a query would require calling job.getQueryResults() in order to determine when the job is finished creating the table, and would then be confusing because that method also returns the rows.

Comment (steffnay) [reply samples/browseTable.js:41]: Oh, meant to also add that row values are accessible in the same way that would be expected with objects in Node.js, so I think we'd be safe in opting to not explicitly show it here.

Comment (tswast) [reply samples/browseTable.js:41]: Hmmm... I don't like that it's not easier to download rows from across projects, but that shouldn't block this sample.

> job.getQueryResults() in order to determine when the job is finished creating the table, and would then be confusing because that method also returns the rows.

See my other comment. I have a concern if `job.getQueryResults()` downloads all rows. The main point of pagination is to save on memory resources. It's bad if it's not possible to do so with this client for query results. (Maybe there's a feature request hidden in here for a getQueryResultsStream method?)

File: samples/queryPagination.js
Comment (steffnay) [reply samples/queryPagination.js]: Yeah, updated.

File: samples/browseTable.js
Comment (steffnay) [reply samples/browseTable.js]: You can paginate. I was just thinking that the reason for using `job.getQueryResults()` in this way might not be clear to users, but I think you're right that it's better to do it that way. Updated.

**Review by tswast — APPROVED** [review]

**Review by JustinBeckwith — APPROVED** [review]

**Comment by tmatsuo** [comment]
Re: snippet-bot comment

Seems like it's going to delete a region tag which is used in devsite. It is ok to merge if you plan to update the devsite usage to the new location soon.

If you think it's a false positive, let me know.

**Comment by tmatsuo** [comment]
However, the correct course of action from the Drift team's recommendation is to 1) first duplicate the region tag, 2) update devsite, then 3) remove the old region tag.

**Comment by steffnay** [comment]
@tmatsuo My bad, will follow that process from now on! I'm waiting for devsite update to be reviewed, so should land soon.

**Pull request merged**
